src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationScopeParsingTests.cs

```diff
@@ -3692,25 +3682,6 @@ public void DeclExpr_03(LanguageVersion langVersion)
                                 N(SyntaxKind.IdentifierToken, "var");
                             }
                         }
-                        M(SyntaxKind.SemicolonToken);
-                    }
-                }
-                N(SyntaxKind.GlobalStatement);
-                {
-                    N(SyntaxKind.ExpressionStatement);
-                    {
-                        N(SyntaxKind.SimpleAssignmentExpression);
-                        {
-                            M(SyntaxKind.IdentifierName);
-                            {
-                                M(SyntaxKind.IdentifierToken);
-                            }
-                            N(SyntaxKind.EqualsToken);
-                            N(SyntaxKind.IdentifierName);
-                            {
-                                N(SyntaxKind.IdentifierToken, "M");
-                            }
-                        }
```

Comment:
old code basically tried to guess what was going on after the `)`, getting is quite wrong.  new paths basically go into error recovery and state that we can't handle things until we get to the next semicolon.

Reply:
this also means a lot less broken constructs and a lot less error messages in these bogus cases.